### DISCUSSION TITLE
How we can configure speechly in Next js project?

### DISCUSSION THREAD

**Discussion by Sajid-Hussain-Nagari** [speechly/speechly#172] [Q&A] [answered]
Hello speechly and web  community,
I am facing an issue during speechly configuration with next js project. I have install two dependencies which are following with their version.
1. "@speechly/react-client": "^2.0.1",
2.  "@speechly/react-ui": "^2.6.1",

when I run my project on localhost it runs successfully but when I am try to record something with speechly mic it popup 
**Unhandled Runtime Error
Error: No Speechly client (are you using Speechly in non-browser environment?)**

I have added my appId with speechprovider in _app.tsx file as follow and I had check with speechProvider appId put in head as well as Inside reduxProvider but facing same issue.

`    <CacheProvider value={emotionCache}>
      <Head>
        <title>NextJS Typescript Admin Dashboard</title>
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1, shrink-to-fit=no"
        />
      </Head>
      <ReduxProvider store={store}>
      <SpeechProvider appId="xxxxxx-f855-4fb2-xxx-xxxxxxxxxxxx"></SpeechProvider>
      </ReduxProvider>
    </CacheProvider>`

and added this code in other page where I wanna use speechly
`      <BigTranscript placement="top" />
      <PushToTalkButton placement="bottom" captureKey=" " powerOn="auto" />
      <IntroPopup />`

kindly suggest me solution for this issue
Thanks

**Comment by arzga** [comment]
Hi Sajid,

I believe you've encountered a bug with the new release of SpeechlyClient. We'll work on the patch and release it shortly. In the meantime, you can try with an earlier version of SpeechlyClient. Thanks for letting us know!

**Comment by arzga** [comment]
I'm finally looking at the issue. However, I haven't yet been able to reproduce the issue in a simple app created with `yarn create next-app --typescript`. I'll attach the `package.json` here. Please let me know if there's major differences in your setup.

**Reply by arzga** [reply]
```
{
  "name": "@speechly/next-js-example",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "next": "12.1.4",
    "react": ">=16.13.1",
    "react-dom": ">=16.13.1",
    "@speechly/react-client": "~2.0.1",
    "@speechly/react-ui": "~2.6.1"
  },
  "devDependencies": {
    "@types/node": ">=16.3.2",
    "@types/react": ">=16.13.1",
    "@types/react-dom": ">=16.13.1",
    "eslint": "8.12.0",
    "eslint-config-next": "12.1.4",
    "typescript": "4.6.3"
  }
}
```

**Reply by arzga** [reply]
The above setup runs Speechly React components (SpeechProvider, PushToTalkButton, TranscriptDrawer, IntroPopup) without problems with them placed in my `_app.tsx`. They also work with static HTML export so I'm still a bit puzzled with what might be causing the problems.

Which version of React are you on? Are you using any special features Next.js features that I should test with?

I'll be adding my Next.js example to the repository shortly and continue the tests with newer React versions.

**Comment by arzga** [comment, marked as answer]
There is now a `/examples/next-js-example/` in this monorepository. It's a two-page Next.js site bootstrapped with `yarn create next-app --typescript` with Speechly UI components on them.

There's instructions on how to use the Rush monorepo on https://github.com/speechly/speechly#how-to-use-this-rush-monorepository
